R/verb-expand.R

```diff
@@ -0,0 +1,44 @@
+#' @importFrom tidyr expand
+#' @export
+expand.tbl_lazy <- function(data, ..., .name_repair = "check_unique") {
+  # TODO sort output?
+  # TODO wait for bugfix: distinct() ignores groups
+  # https://github.com/tidyverse/dbplyr/issues/535
+  dots <- quos_auto_name(quos(...))
+  dots <- purrr::discard(dots, rlang::quo_is_null)
+  names(dots) <- vctrs::vec_as_names(names(dots), repair = .name_repair)
```

Comment:
I _think_ this should just be left to `distinct()` to handle

Reply:
This doesn't work for two reasons:
1. `distinct()` doesn't rename with a repair strategy
2. `distinct()` cannot handle this as the code iterates over `dots` and therefore `distinct()` never sees all the variables at once.

Reply:
In that case, I'd suggest filing an issue to remind us that distinct needs to duplicated variables, and we can come back to it later.

Reply:
Hmm, seems like we misunderstand each other. `distinct()` does handle duplicated variables. It processes them step by step, similar to what `mutate()` does. See the following example.

    library(dplyr)
    library(dbplyr)

    memdb_frame(x = 1) %>% 
      distinct(x = x + 1, x = x + 1)
    #> # Source:   lazy query [?? x 1]
    #> # Database: sqlite 3.33.0 [:memory:]
    #>       x
    #>   <dbl>
    #> 1     3

    tibble(x = 1) %>% 
      distinct(x = x + 1, x = x + 1)
    #> # A tibble: 1 x 1
    #>       x
    #>   <dbl>
    #> 1     3

<sup>Created on 2020-11-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

In comparison `tidyr::expand()` renames

    tidyr::expand(
      dplyr::tibble(x = 1:2),
      x, x,
      .name_repair = "unique"
    )
    #> New names:
    #> * x -> x...1
    #> * x -> x...2
    #> # A tibble: 4 x 2
    #>   x...1 x...2
    #>   <int> <int>
    #> 1     1     1
    #> 2     1     2
    #> 3     2     1
    #> 4     2     2

<sup>Created on 2020-11-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

So, I don't see how we can use `distinct()` to handle the renaming. FYI I had to update my strategy for renaming to also handle variables in `nesting()`

Reply:
Oooooook, now I get it!